content/blog/2018/istio-authorization/index.md

```diff
@@ -0,0 +1,211 @@
+authorization decision is completely done locally inside an Envoy filter, without dependency to any external module.
+This allows Istio authorization to achieve high performance and availability.
+
+### Work with/without primary identities
+
+Like any other RBAC system, Istio authorization is identity aware. In Istio authorization policy, there is a primary
+identity called `user`, which represents the principal of the client.
+
+In addition to the primary identity, you can also specify any conditions that define the identities. For example,
+you can specify the client identity as “user Alice calling from Bookstore frontend service”, in which case,
+you have a combined identity of the calling service (`Bookstore frontend`) and the end user (`Alice`).
+
+To achieve best security, it is strongly recommended that users enable [authentication features](/docs/concepts/security/#authentication),
+and use authenticated identities in authorization policies. However, strongly authenticated identity is not required
+for using authorization. Istio authorization works with or without identities. If you are working with a legacy system,
+you may not have mutual TLS or JWT authentication setup for your mesh. In this case, the only way to identify the client is, say,
+through IP. You can still use Istio authorization to control which IP addresses or IP ranges are allowed to access your service.
+
+## Examples
+
+In [authorization task page](/docs/tasks/security/role-based-access-control/), you have already learned how to
+use Istio's authorization feature to control namespace level and service level access using the example of
+[BookInfo application](/docs/examples/bookinfo/). In this section, you'll see more examples on how to achieve
+micro-segmentation with Istio authorization.
+
+### Namespace level segmentation via RBAC + conditions
+
+Suppose you have services in `frontend` namespace and `backend` namespace. You would like to allow all your services
+in `frontend` namespace to access all services that are marked `external` in `backend` namespace.
+
+{{< text yaml >}}
+apiVersion: "rbac.istio.io/v1alpha1"
+kind: ServiceRole
+metadata:
+  name: external-api-caller
+  namespace: backend
+spec:
+  rules:
+  - services: ["*"]
+    methods: ["*”]
+    constraints:
+    - key: “destination.labels[visibility]”
+      values: ["external"]
+---
+apiVersion: "rbac.istio.io/v1alpha1"
+kind: ServiceRoleBinding
+metadata:
+  name: external-api-caller
+  namespace: backend
+spec:
+  subjects:
+  - properties:
+      source.namespace: “frontend”
+  roleRef:
+    kind: ServiceRole
+    name: "external-api-caller"
+{{< /text >}}
+
+The `ServiceRole` and `ServiceRoleBinding` above expressed “*who* is allowed to do *what* under *which conditions*”
+(RBAC + conditions). Specifically,
+
+* **“who”** is services in `frontend` namespace.
+* **“what”** is to call services in `backend` namespace.
+* **“conditions”** is `visibility` label of the destination service is `external`.
```

Comment:
is the `visibility` label of the destination service having the value `external`.